### PR TITLE
[PFS-207] Add GetWithID Functions to Projects and Repos

### PR DESCRIPTION
**Pull request by FahadBSyed**
The goal of this PR is to add `GetXWithID` functions to Projects and Repos so they can be consumed by PFS APIs.

### CODE REVIEW

File: src/internal/pfsdb/projects.go [hunk 1]
```diff
@@ -37,8 +37,7 @@ func (err *ProjectNotFoundError) Error() string {
 }
 
 func (err *ProjectNotFoundError) Is(other error) bool {
-	_, ok := other.(*ProjectNotFoundError)
-	return ok
+	return errors.As(other, &ProjectNotFoundError{})
```
Comment (brendoncarroll) [inline src/internal/pfsdb/projects.go:40]: Check if you need the double pointer here.

File: src/internal/pfsdb/projects_test.go [hunk 2]
```diff
@@ -79,13 +79,20 @@ func TestGetProject(t *testing.T) {
...
 		require.Equal(t, createInfo.Description, getInfo.Description)
+		// validate GetProjectWithID.
+		getInfoWithID, err := pfsdb.GetProjectWithID(cbCtx, tx, testProj)
+		require.NoError(t, err, "should be able to get a project")
+		require.Equal(t, createInfo.Project.Name, getInfoWithID.ProjectInfo.Project.Name)
+		require.Equal(t, createInfo.Description, getInfoWithID.ProjectInfo.Description)
+		// validate error for attempting to get non-existent project.
 		_, err = pfsdb.GetProject(cbCtx, tx, 3)
 		require.YesError(t, err, "should not be able to get non-existent project")
+		fmt.Println(err)
```
Comment (brendoncarroll) [inline src/internal/pfsdb/projects_test.go:95]: Remove

Comment (brendoncarroll) [reply src/internal/pfsdb/projects_test.go:95]: Or change it to log with `t.Log` if you want to keep it.

**Review by brendoncarroll — APPROVED** [review]
A few things.  Otherwise LGTM.

`errors.As` is tricky.  I would exercise that code path in a test.

File: src/internal/pfsdb/projects_test.go (hunk 2 above)
Comment (FahadBSyed) [reply src/internal/pfsdb/projects_test.go:95]: whoops! Nice catch

File: src/internal/pfsdb/projects.go (hunk 1 above)
Comment (FahadBSyed) [reply src/internal/pfsdb/projects.go:40]: This is actually only one pointer:
``` go
// ProjectNotFoundError is returned by GetProject() when a project is not found in postgres.
type ProjectNotFoundError struct {
	Name string
	ID   ProjectID
}
```

But to confirm, that this works, we're using it in `projects_test.go`
```
...
		// validate error for attempting to get non-existent project.
		_, err = pfsdb.GetProject(cbCtx, tx, 3)
		require.YesError(t, err, "should not be able to get non-existent project")
		require.True(t, (&pfsdb.ProjectNotFoundError{ID: 3}).Is(err))
		return nil
	}))
```

**Pull request merged**